src/restLink.ts

```diff
@@ -1042,10 +1042,12 @@ const resolver: Resolver = async (
     );
   }
 
-  if (endpointOption.responseTransformer) {
-    result = endpointOption.responseTransformer(result, type);
-  } else if (responseTransformer) {
-    result = responseTransformer(result, type);
+  const transformer = endpointOption.responseTransformer || responseTransformer;
+  if (transformer) {
+    // A responseTransformer might call something else than json() on the Body.
+    result = await transformer(result, type);
```

Comment:
More of a question for the group, now that I am looking at this, but since this is going out into user land for a (potentially) async operation, should we be wrapping and rethrowing this error with some more meaningful context?

I'm just curious what happens if this user provided function throws.